packages/arb-rpc-node/dev/ethcall_test.go

```diff
@@ -0,0 +1,180 @@
+package dev
+
+import (
+	"encoding/hex"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/offchainlabs/arbitrum/packages/arb-evm/arbos"
+	"github.com/offchainlabs/arbitrum/packages/arb-evm/arboscontracts"
+	"github.com/offchainlabs/arbitrum/packages/arb-rpc-node/arbostestcontracts"
+	"github.com/offchainlabs/arbitrum/packages/arb-rpc-node/web3"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/common"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/protocol"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/test"
+)
+
+func expectHex(t *testing.T, result []byte, resultErr error, expected string) {
+	test.FailIfError(t, resultErr)
+	if ethcommon.BytesToHash(result) != ethcommon.HexToHash(expected) {
+		t.Fatal("Unexpected ", expected, "Got", ethcommon.BytesToHash(result))
+	}
```

Comment:
I think this only checks the first 32 bytes. It'd be safer to either do a `bytes.Equal` on two byte slices or do a length check on the inputs to make sure they're the expected length